romancal/outlier_detection/tests/test_outlier_detection.py

```diff
@@ -1,42 +1,358 @@
+                            "exptype": "science"
+                        },
+                        {
+                            "expname": "img_2.asdf",
+                            "exptype": "science"
+                        }
+                    ]
+                }
+            ]
+        }
+    """
+    if members_mapping is not None:
+        asn_dict = json.loads(asn_content)
+        asn_dict["products"][0]["members"] = []
+        for x in members_mapping:
+            asn_dict["products"][0]["members"].append(x)
+        asn_content = json.dumps(asn_dict)
+
+    asn_file_path = str(tmp_path / "sample_asn.json")
+    asn_file = StringIO()
+    asn_file.write(asn_content)
+    with open(asn_file_path, mode="w") as f:
+        print(asn_file.getvalue(), file=f)
+
+    return asn_file_path
+
+
+@pytest.mark.parametrize(
+    "input_models",
+    [
+        list(),
+        [""],
+        "",
+        "testing",
+        {},
+    ],
 )
+def test_outlier_raises_error_on_invalid_input_models(input_models):
+    """Test that OutlierDetection raises an error if input is not a ModelContainer."""
+
+    step = OutlierDetectionStep()
+
+    with pytest.raises(Exception) as exec_info:
+        step.process(input_models)
+
+    assert exec_info.type == TypeError
+
+
+def test_outlier_skips_step_on_invalid_number_of_elements_in_input(base_image):
+    img = base_image()
+
+    step = OutlierDetectionStep()
+    step.input_models = ModelContainer([img])
+
+    step._run_process()
+
+    assert step.input_models[0].meta.cal_step.outlier_detection == "SKIPPED"
+    assert step.skip
+
+
+def test_outlier_skips_step_on_exposure_type_different_from_wfi_image(base_image):
+    img_1 = base_image()
+    img_1.meta.exposure.type = "WFI_PRISM"
+    img_2 = base_image()
+    img_2.meta.exposure.type = "WFI_PRISM"
+
+    step = OutlierDetectionStep()
+    step.input_models = ModelContainer([img_1, img_2])
+
+    step._run_process()
+
+    assert step.input_models[0].meta.cal_step.outlier_detection == "SKIPPED"
+    assert step.skip
+
+
+def test_outlier_valid_input_datamodels(tmp_path, base_image):
+    img_1 = base_image()
+    img_1.meta.filename = "img_1.asdf"
+    img_2 = base_image()
+    img_2.meta.filename = "img_2.asdf"
+
+    step = OutlierDetectionStep()
+    step.input_models = ModelContainer([img_1, img_2])
+
+    step._run_process()
+
+    assert step.skip is False
+    assert img_1.meta.cal_step.outlier_detection == "COMPLETE"
+    assert img_2.meta.cal_step.outlier_detection == "COMPLETE"
+
+
+def test_outlier_valid_input_asn(tmp_path, base_image):
+    img_1 = base_image()
+    img_1.meta.filename = "img_1.asdf"
+    img_1.save(tmp_path / "img_1.asdf")
+    img_2 = base_image()
+    img_1.meta.filename = "img_2.asdf"
+    img_2.save(tmp_path / "img_2.asdf")
+
+    asn_filepath = create_asn_file(tmp_path)
+
+    step = OutlierDetectionStep()
+    step.input_models = ModelContainer(asn_filepath)
```

Comment:
Is `outlier_detection` supposed to run with associations as input?
If so the handling of the association should be done within the `process` function.